### DISCUSSION TITLE
Windows install error

### DISCUSSION THREAD

**Discussion by sophielamb** [thegetty/quire#16] [Get Help] [answered]
Hello,
I've been having difficulty getting Quire set-up on my Windows laptop. Following the guidelines I successfully installed Node.Js, Pandoc, Prince XML and Windows Build Tools. I'm pretty sure they are all there as they appear in my explorer. 

However, I tried to pull Quire from the repo. At first appeared to have installed, but I double checking using the --version command, and then --help command. Powershell tells me that it cannot be loaded as it is disabled. 

The error provided a link to the documentation for Win PowerShell execution policies. My laptop is setup with the default Restricted policy, which prevents running of all script files, including formatting and configuration files (.ps1xml), module script files (.psm1), and PowerShell profiles (.ps1). Do I need to change the restriction to allow the configuration files to run?

<br>

This is the text from PowerShell (my commands are in bold):

<br>

PS C:\Users\sophi> **npm install --global @thegetty/quire-cli**

npm WARN deprecated axios@0.19.2: Critical security vulnerability fixed in v0.21.1. For more information, see https://github.com/axios/axios/pull/3410

npm WARN deprecated cheerio-select-tmp@0.1.1: Use cheerio-select instead

npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated

npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.

npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.

C:\Users\sophi\AppData\Roaming\npm\quire -> C:\Users\sophi\AppData\Roaming\npm\node_modules\@thegetty\quire-cli\bin\index.js

 

> core-js@3.9.0 postinstall C:\Users\sophi\AppData\Roaming\npm\node_modules\@thegetty\quire-cli\node_modules\core-js

> node -e "try{require('./postinstall')}catch(e){}"

 

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

 

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon:

> https://opencollective.com/core-js

> https://www.patreon.com/zloirock

 

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)

 

 

> hugo-bin@0.49.0 postinstall C:\Users\sophi\AppData\Roaming\npm\node_modules\@thegetty\quire-cli\node_modules\hugo-bin

> rimraf vendor && node lib/install

 

√  success   Hugo binary installed successfully!

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.3.1(node_modules\@thegetty\quire-cli\node_modules\chokidar\node_modules\fsevents):

npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7(node_modules\@thegetty\quire-cli\node_modules\watchpack-chokidar2\node_modules\chokidar\node_modules\fsevents):

npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

 

+ @thegetty/quire-cli@0.18.2

added 591 packages from 284 contributors in 59.675s

PS C:\Users\sophi> **quire --version**

quire : File C:\Users\sophi\AppData\Roaming\npm\quire.ps1 cannot be loaded because running scripts

is disabled on this system. For more information, see about_Execution_Policies at

https:/go.microsoft.com/fwlink/?LinkID=135170.

At line:1 char:1

+ quire --version

+ ~~~~~

    + CategoryInfo          : SecurityError: (:) [], PSSecurityException

    + FullyQualifiedErrorId : UnauthorizedAccess

PS C:\Users\sophi> **quire --help**

quire : File C:\Users\sophi\AppData\Roaming\npm\quire.ps1 cannot be loaded because running scripts

is disabled on this system. For more information, see about_Execution_Policies at

https:/go.microsoft.com/fwlink/?LinkID=135170.

At line:1 char:1

+ quire --help

+ ~~~~~

    + CategoryInfo          : SecurityError: (:) [], PSSecurityException

    + FullyQualifiedErrorId : UnauthorizedAccess

PS C:\Users\sophi>

**Comment by thom4parisot** [comment, marked as answer]
I don't remember facing this issue on Windows, but I spotted this bit of information in the (looong) error trace:

> quire.ps1 cannot be loaded because running scripts is disabled on this system. For more information, see about_Execution_Policies at https:/go.microsoft.com/fwlink/?LinkID=135170.

If you visit https:/go.microsoft.com/fwlink/?LinkID=135170, do you find any piece of information that makes sense to you?

**Reply by sophielamb** [reply]
I did see the link provided - I should have mentioned that - it goes to PowerShell documentation about execution policies. 

I got the execution policy list for my machine, it appears that my laptop has the default Restricted policy. 

Will changing it to something like _RemoteSigned_ allow the full Quire install? I don't want to change the restriction settings without advice, since I really don't want to end up doing something dangerous with the security.
<br>
edit:
<br>
I misread my powershell output - the scope is currently undefined (see below). I'm not certain what would be a suitable choice of policy. Alternatively, a stack overflow answer on policies suggested using -ExecutionPolicy Bypass ,could that work?
<br>
~~~~~~~~
PS C:\WINDOWS\system32> Get-ExecutionPolicy -List

        Scope ExecutionPolicy
        ----- ---------------
MachinePolicy       Undefined
   UserPolicy       Undefined
      Process       Undefined
  CurrentUser       Undefined
 LocalMachine       Undefined

**Reply by sophielamb** [reply]
I have now changed the execution policy for CurrentUser to RemoteSigned , as the the documentation says it's the default for Win server computers. This has worked and I now have access to Quire, so that's been fixed I recken.

**Reply by Erin-Cecele** [reply]
@sophielamb That is wonderful news! Glad to hear you were able to get up-and-running.
